ui/src/app/base/components/TitledSection/TitledSection.tsx

```diff
@@ -0,0 +1,34 @@
+import type { ReactNode } from "react";
+
+import { Strip } from "@canonical/react-components";
+import type { PropsWithSpread, StripProps } from "@canonical/react-components";
+
+import { useId } from "app/base/hooks/base";
+
+type Props = PropsWithSpread<
+  {
+    children?: ReactNode;
+    title: ReactNode;
+  },
+  StripProps
+>;
+
+const TitledSection = ({ children, title, ...props }: Props): JSX.Element => {
+  const id = useId();
+  return (
+    <Strip
+      shallow
+      element="section"
+      aria-labelledby={id}
+      data-testid="titled-section"
+      {...props}
+    >
+      <h2 id={id} className="p-heading--4">
+        {title}
+      </h2>
+      {children}
```

Comment:
Some of the sections will need a button next to the title. Usually I would handle this by doing
``` jsx
<div className="u-flex--between">
  <h2>Title</h2>
  <Button>Button</Button>
</div>
```

Can this be amended to make that possible?

Reply:
Done.